editor/inserter/menu.js

```diff
@@ -58,37 +56,66 @@ class InserterMenu extends Component {
 		} );
 	}
 
-	selectBlock( name ) {
+	selectBlock( blockType ) {
 		return () => {
-			this.props.onSelect( name );
+			this.props.onSelect( blockType.name );
+			recordUsage( blockType );
```

Comment:
Sounds good. Though it'd be fine to do the REST API pieces separately, as it's not immediately obvious where this should be stored.